krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -127,8 +130,12 @@ CompletableFuture<NodeResponse> executeRequestCommand(NodeRequestCommand nodeCom
     try {
       boolean executeMainLogic;
       if (nodeCommand instanceof SkipNode skipNode) {
-        resultForRequest.completeExceptionally(skipNodeException(skipNode));
-        return resultForRequest;
+        requestsByDependantChain
+            .computeIfAbsent(skipNode.dependentChain(), k -> new LinkedHashSet<>())
+            .add(requestId);
+        dependantChainByRequest.computeIfAbsent(requestId, r -> skipNode.dependentChain());
```

Comment:
for such use cases, `putIfAbsent` can be used since no new object creation or complex computation is happening in the computation logic.

In this particular case, just `put` will be better since we don't expect to get multiple dependantChains for the same requestId

Performance of put > putIfAbsent > computeIfAbsent

Reply:
but we have it as a set of request ids, so why cant we have multiple requestIds for a dependent chain
This is the code we are using for executeWithInputs. 
else if (nodeCommand instanceof ExecuteWithInputs executeWithInputs) {
        requestsByDependantChain
            .computeIfAbsent(executeWithInputs.dependantChain(), k -> new LinkedHashSet<>())
            .add(requestId);
}